app/components/scheduled_post_indicator/index.ts

```diff
@@ -2,23 +2,35 @@
 // See LICENSE.txt for license information.
 
 import {withDatabase, withObservables} from '@nozbe/watermelondb/react';
-import {map} from 'rxjs/operators';
+import {map, switchMap} from 'rxjs/operators';
 
 import {getDisplayNamePreferenceAsBool} from '@helpers/api/preference';
 import {queryDisplayNamePreferences} from '@queries/servers/preference';
+import {observeScheduledPostCountForChannel} from '@queries/servers/scheduled_post';
+import {observeCurrentTeamId} from '@queries/servers/system';
 import {observeCurrentUser} from '@queries/servers/user';
 
 import {ScheduledPostIndicator} from './scheduled_post_indicator';
 
-const enhance = withObservables([], ({database}) => {
+import type {WithDatabaseArgs} from '@typings/database/database';
+
+type Props = WithDatabaseArgs & {
+    channelId: string;
+}
+
+const enhance = withObservables(['channelId'], ({database, channelId}: Props) => {
     const currentUser = observeCurrentUser(database);
+    const currentTeamId = observeCurrentTeamId(database);
     const preferences = queryDisplayNamePreferences(database).
         observeWithColumns(['value']);
     const isMilitaryTime = preferences.pipe(map((prefs) => getDisplayNamePreferenceAsBool(prefs, 'use_military_time')));
 
+    // TODO: fix the count for DM's and GM's and Thread
+    const scheduledPostCount = currentTeamId.pipe(switchMap((teamId) => observeScheduledPostCountForChannel(database, teamId, channelId)));
```

Comment:
1. We need to include thread ID as well (root post's ID) as when in a thread, we need to show only that thread's count. Similarly when in channels, we need to exclude thread scheduled posts from the count.

2. You also need to fetch the scheduled post if count is `1`, and replace the hardcoded value in the component where there's a todo `//TODO: remove this hardcoded value with actual value` with the scheduled post's scheduled at time. We need to display that time if there is only a single scheduled post in channel/thread.

Reply:
Only when crt is enabled, correct?

Reply:
Yup fixed it.